src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -231,13 +238,28 @@ private void fileAclOp(AclOp op, Dataset dataset, List<String> fileIds, String r
         Acl.Group readerGroup = new Acl.Group(readersPolicyEmail);
         Acl acl = Acl.newBuilder(readerGroup, Acl.Role.READER).build();
 
-        for (String fileId : fileIds) {
-            FSItem fsItem = fileDao.retrieveById(dataset, fileId, 0, true);
-            if (fsItem instanceof FSFile) {
-                FSFile fsFile = (FSFile) fsItem;
-                GoogleBucketResource bucketForFile = dataLocationService.lookupBucket(fsFile.getBucketResourceId());
+        Map<String, GoogleBucketResource> bucketCache = new HashMap<>();
+
+        int batchSize = configurationService.getParameterValue(FIRESTORE_SNAPSHOT_BATCH_SIZE);
+        List<List<String>> batches = ListUtils.partition(fileIds, batchSize);
```

Comment:
Good question. Should not double the memory, since the objects are not copied.
It is ok with you, I will make another ticket to review the use of the partition.

We should probably revisit the format of our file ids. We might get more memory benefit if we used the base64 encoding version

Reply:
Hrm, I played around with the guava version of partition and it did substantially increase memory usage when the partitioned object was used. Perhaps the Apache commons collection one is smarter?

Reply:
I looked at the code. Is very clever. It is a list of sublist() over the original list. So there actually is no new list at all. It is all computations on top of the original list. It should be no memory impact worth talking about